test/general/data-structures/vector_test.c

```diff
@@ -0,0 +1,139 @@
+#include <stdlib.h>
+#include <stdio.h>
+#include "core/utils/vector.h"
```

Comment:
Maybe the directory of this test file should mirror the path of the code under test (i.e., `utils` instead of `data-structures`)?

Reply:
Sounds good -- the change has been applied.